spec/models/podcast_episode_appearance_spec.rb

```diff
@@ -0,0 +1,20 @@
+require "rails_helper"
+
+RSpec.describe PodcastEpisodeAppearance, type: :model do
+  let(:podcast_episode_appearance) { create(:podcast_episode_appearance) }
+
+  describe "validations" do
+    subject { podcast_episode_appearance }
+
+    it do
+      expect(podcast_episode_appearance).to belong_to(:user).class_name("User").inverse_of(:podcast_episode_appearances)
+        .inverse_of(:podcast_episode_appearances)
+    end
+
+    it { is_expected.to belong_to(:podcast_episode) }
+
+    it { is_expected.to validate_presence_of(:podcast_episode_id) }
+    it { is_expected.to validate_presence_of(:user_id) }
```

Comment:
missing the presence check and inclusion validation for `role`

Reply:
@rhymes Can you give me a hand? I'm constantly getting:
`# PodcastEpisodeAppearance validations is expected to validate that :role cannot be empty/falsy`, even though I've added ` role { "guest" }` into factory bot class. It returns:
```ArgumentError: invalid value for Integer(): "". ```
 I've also tried `  let(:podcast_episode_appearance) { create(:podcast_episode_appearance, role: "guest") }` but it gives the same result. I've checked - factory bot gives me a guest value for the role of instance it created.

Reply:
I found the solution, it was a weird error message indeed. It doesn't have to do with `shoulda-matchers` but I was able to reproduce it locally in the console:

```ruby
> pea = PodcastEpisodeAppearance.new
> pea.role = "foobar"
> pea.validate!
ArgumentError: invalid value for Integer(): "foobar"
from /forem/vendor/bundle/ruby/2.7.0/gems/i18n-1.8.5/lib/i18n/interpolate/ruby.rb:34:in `sprintf'
```

Basically the issue is the error message that's lacking a `s` for the string interpolation.

It should be:

```ruby
validates :role, inclusion: { in: %w[host guest], message: "%<value>s is not a valid role" }
```

and not

```ruby
validates :role, inclusion: { in: %w[host guest], message: "%<value> is not a valid role" }
```

Reply:
Really weird. I google so many things about `shoulda-matchers` and inclusion validation. I've even tested many things about FactoryBot instance creation, and at the end, the error is so silly. It's not verbose and descriptive at all. Thanks for your help :clap: